collect-intent.rst

```diff
@@ -0,0 +1,207 @@
+
+.. _use-intent:
+
+Using Collect's Intent
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each screen of an Android app is called an activity. ODK Collect has several activities and sub-activities. These activities are able to communicate with each other using intents.
+
+If you want to start ODK Collect's activity you need to:
+
+1. Create a new intent using an appropriate action.
+2. Set the type of created intent.
+3. Start an activity using the intent.
+
+.. _edit-form:
+
+Navigating user to choose and edit a form/instance 
+""""""""""""""""""""""""""""""""""""""""""""""""""""
+ 
+.. code-block:: java
+ 	
+  Intent intent = new Intent(android.intent.action.EDIT);
+  intent.setData(org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.CONTENT_URI);
+ 
+This will allow user to choose the form and go on filling it .
+ 
+Similarly for an instance of the form : 
+ 
+.. code-block:: java
+ 
+  Intent intent = new Intent(android.intent.action.EDIT);
+  intent.setData(org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.CONTENT_URI);
+
+.. _get-uri: 	
+ 
+Getting the URI of the form/instance chosen by user
+"""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+.. _start-activity:
+
+Starting Activity For Result
+''''''''''''''''''''''''''''''
+
+.. code-block:: java
+ 
+  Intent intent = new Intent(android.intent.action.PICK);
+  intent.setData(org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.CONTENT_URI);
+ 
+.. code-block:: java
+ 
+  static final int PICK_FORM_REQUEST = 1;  // The request code
+  startActivityForResult(intent, PICK_FORM_REQUEST);
+
+.. _get-result:
+
+Getting result
+''''''''''''''''
+ 
+To get the result, override ``onActivityResultMethod`` in the followig way:
+
+.. code-block:: java
+
+  @Override
+  protected void onActivityResult(int requestCode, int resultCode, Intent formUri) {
+     // Check which request we're responding to
+    if (requestCode == PICK_FORM_REQUEST) {
+        // Make sure the request was successful
+	    if (resultCode == RESULT_OK) {
+ 	      // The user picked a contact.
+ 	      // The Intent's data Uri identifies which form was selected.
+ 	      // Do something with the form here
+ 	    }
+	}	
+  }
+ 
+
+Similarly for an instance, change the URI to that of the instance :
+ 
+.. code-block:: java
+ 
+  intent.setData(org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.CONTENT_URI);
+
+.. _use-form-uri:
+
+Use form's URI to edit/view form
+""""""""""""""""""""""""""""""""""
+ 
+The formURI in the ``onActivityResult()`` method, allows us to view/edit the particular form by:
+ 
+.. code-block:: java
+ 
+  Intent intent = new Intent(android.intent.action.EDIT);
+  intent.setData(formUri);
+ 
+If we want to view the form, the action can be changed to :
+ 
+.. code-block:: java
+ 
+  Intent intent = new Intent(android.intent.action.VIEW);
+ 
+Similar things can be done for an Instance.
+ 
+.. note::
```

Comment:
I would move this note up to right after the introductory "understanding intents" section. I think for most people going straight to an example would be most helpful.